File: Compiler/UnitTestSample_1/TryCatch.cs
using JavaScript;

namespace UnitTestSample_1
{
    class TryCatch
    {
        private void DoSomething()
        {
            throw new Error("err");
        }

        private void DoClean()
        {
        }

        private void DoHandle(object obj)
        {
        }

        public void TC()
        {
            try
            {
                DoSomething();
            }
            catch (Error e)
            {
                DoHandle(e);
            }
        }

        public void TC1()
        {
            try
            {
                DoSomething();
            }
            catch
            {
                DoHandle(null);
            }
        }

        public void TF()
        {
            try
            {
                DoSomething();
            }
            finally
            {
                this.DoClean();
            }
        }

        public void TCF()
        {
            try
            {
                DoSomething();
            }
            catch (EvalError e)
            {
                DoHandle(e);
            }
            finally
            {
                DoClean();
            }
        }

        public void MultiCatch()
        {
            try
            {
                DoSomething();
            }
            catch (EvalError e)
            {
                DoHandle(e);
            }
            catch (RangeError e)
            {
                DoHandle(e);
            }
            catch
            {
                DoHandle(null);
            }
            finally
            {
                DoClean();
            }
        }
    }
}

File: Compiler/UnitTestSample_1/ToCompare/TryCatch.js
enyo.kind({
    name: "UnitTestSample_1.TryCatch",
    kind: "enyo.Object",

    doSomething : function(){
        throw new Error("err");
    },

    doClean : function(){
    },

    doHandle : function(obj){
    },

    tC : function(){
        try {
            this.doSomething();
        } catch (e) {
            this.doHandle(e);
        }
    },

    tC1 : function(){
        try {
            this.doSomething();
        } catch (e) {
            this.doHandle(null);
        }
    },

    tF : function(){
        try {
            this.doSomething();
        } finally {
            this.doClean();
        }
    },

    tCF : function(){
        try {
            this.doSomething();
        } catch (e) {
            this.doHandle(e);
        } finally {
            this.doClean();
        }
    },

    multiCatch : function(){
        try {
            this.doSomething();
        } catch (e) {
            if (e.name == "EvalError") {
                this.doHandle(e);
            } else if (e.name == "RangeError") {
                this.doHandle(e);
            } else {
                this.doHandle(null);
            }
        } finally {
            this.doClean();
        }
    }
});
